fastcrypto-zkp/src/bn254/zk_login_api.rs

```diff
@@ -169,3 +170,27 @@ fn verify_zk_login_proof_with_fixed_vk(
     Groth16::<Bn254>::verify_with_processed_vk(&GLOBAL_VERIFYING_KEY, public_inputs, proof)
         .map_err(|e| FastCryptoError::GeneralError(e.to_string()))
 }
+
+/// Verify that the given parameters (name, value, aud, iss and salt_hash) were used to generate the
+/// given address.
+pub fn verify_zk_login_id(
+    address: &[u8],
+    name: &str,
+    value: &str,
+    aud: &str,
+    iss: &str,
+    salt_hash: &str,
+) -> FastCryptoResult<bool> {
+    let address_seed = gen_address_seed_with_salt_hash(salt_hash, name, value, aud)?;
+    verify_zk_login_iss(address, &address_seed, iss)
+}
+
+/// Verify that the given parameters (address_seed and iss) were used to generate the given address.
+pub fn verify_zk_login_iss(
```

Comment:
do we need it pub?

Reply:
Yes, since this is also going to be called from sui-framework.